gui/wxpython/vdigit/toolbars.py

```diff
@@ -882,6 +946,43 @@ def StartEditing(self, mapLayer):
             else:
                 return
 
+        # check and clean zero length lines, boundaries
+        zero_length = self._topoZeroLengthCheck(
+            map_name=mapLayer.GetName(),
+        )
+
+        if zero_length is None:
+            return
+        else:
+            ltype = None
+            if zero_length[0] > 0 and zero_length[1] == 0:
+                    ltype = 'line(s)'
+            elif zero_length[1] > 0 and zero_length[0] == 0:
+                    ltype = 'boundary(ies)'
+            elif zero_length[0] > 0 and zero_length[1] > 0:
+                ltype = 'line(s)/boundary(ies)'
+
+            if ltype:
+                clean_dlg = wx.MessageDialog(
+                    self.GetParent(),
+                    message=_(
+                        "The vector map contains zero-length {ltype}, "
+                        "to continue editing, you need to clean them. "
+                        "Do you want to clean them?".format(ltype=ltype)
+                    ),
+                    caption=_('Clean?'),
+                    style=wx.YES_NO | wx.YES_DEFAULT | wx.ICON_QUESTION,
+                )
```

Comment:
Translatable strings can't have sentences constructed from words. You need multiple messages or part of the message which is not a sentence but some `{x}: {y}` type of report or table. There is also a way how to change a message with support of the gettext package based on the number, but I don't know how to easily use it in Python with our suboptimal translation handling (that is `_` added to buildins).